test/create_sg_test.go

```diff
@@ -0,0 +1,108 @@
+package test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/require"
+)
+
+func initTestCases() []SecurityGroupTestCase {
+
+	return []SecurityGroupTestCase{
+		{
+			testName:         "Test1",
+			expectApplyError: false,
+			vars: map[string]interface{}{
+				"ingress_ports":       []string{"80", "443"},
+				"ingress_cidr_blocks": []string{"0.0.0.0/0"},
+				"egress_cidr_blocks":  []string{"0.0.0.0/0"},
+				"sg_name_prefix":      "",
+				"vpc_id":              "",
+			},
```

Comment:
This is a good basic use of the module.

I think a potential second test case would be a case where the rules exceed the `maximum_rules_per_sg`. This is one of the main benefits of this module is that it creates dynamic number of security groups to match the required number of security group rules.

To test this, we may need to add a few more variables (like expected number of security groups created). We'll also want to set `maximum_rules_per_sg` low to keep the example readable.

We could have this be a follow-on, like checking the content of the rules.